dtm0/fop.c

```diff
@@ -21,85 +21,57 @@
 
 
 #define M0_TRACE_SUBSYSTEM M0_TRACE_SUBSYS_DTM0
-#include "lib/trace.h"
-#include "addb2/addb2.h"
-#include "be/dtm0_log.h"
-#include "dtm0/addb2.h"
 #include "dtm0/fop.h"
 #include "dtm0/fop_xc.h"
-#include "dtm0/service.h"
-#include "dtm0/tx_desc.h"
-#include "dtm0/tx_desc_xc.h"
-#include "lib/errno.h"
+#include "dtm0/addb2.h"
+#include "dtm0/drlink.h"       /* m0_dtm0_req_post */
+#include "dtm0/service.h"      /* m0_dtm0_service */
+#include "be/dtm0_log.h"       /* m0_be_dtm0_log_* */
+#include "fop/fom_generic.h"   /* M0_FOPH_* */
 #include "lib/assert.h"
+#include "lib/errno.h"
 #include "lib/memory.h"
-#include "lib/chan.h"
-#include "lib/finject.h"
-#include "lib/time.h"
-#include "lib/misc.h"           /* M0_IN() */
-#include "fop/fop.h"
-#include "fop/fom.h"
-#include "fop/fom_generic.h"
-#include "fop/fop_item_type.h"
-#include "rpc/rpc_opcodes.h"
-#include "rpc/rpc.h"
-#include "rpc/rpclib.h"
-
-static void dtm0_rpc_item_reply_cb(struct m0_rpc_item *item);
-
-/*
-  RPC item operations structures.
- */
-const struct m0_rpc_item_ops dtm0_req_fop_rpc_item_ops = {
-        .rio_replied = dtm0_rpc_item_reply_cb,
-};
+#include "lib/misc.h"          /* M0_IN() */
+#include "lib/trace.h"         /* M0_LOG */
+#include "reqh/reqh.h"         /* reqh::rh_beseg */
+#include "rpc/rpc_opcodes.h"   /* M0_DTM0_REP_OPCODE */
 
 struct m0_fop_type dtm0_req_fop_fopt;
 struct m0_fop_type dtm0_rep_fop_fopt;
 
+/** Structure that describes DTM0 FOM that handles an incoming DTM0 message. */
+struct dtm0_fom {
+	struct m0_fom dtf_fom;
+};
+
 /*
   Fom specific routines for corresponding fops.
  */
-static int dtm0_fom_tick(struct m0_fom *fom);
+static int dtm0_emsg_fom_tick(struct m0_fom *fom);
+static int dtm0_pmsg_fom_tick(struct m0_fom *fom);
 static int dtm0_fom_create(struct m0_fop *fop, struct m0_fom **out,
 			       struct m0_reqh *reqh);
 static void dtm0_fom_fini(struct m0_fom *fom);
 static size_t dtm0_fom_locality(const struct m0_fom *fom);
 
-static const struct m0_fom_ops dtm0_req_fom_ops = {
+static const struct m0_fom_ops dtm0_pmsg_fom_ops = {
 	.fo_fini = dtm0_fom_fini,
-	.fo_tick = dtm0_fom_tick,
+	.fo_tick = dtm0_pmsg_fom_tick,
 	.fo_home_locality = dtm0_fom_locality
 };
 
-extern struct m0_reqh_service_type dtm0_service_type;
-
-enum dtm0_phases {
-	DTM0_REQ_WHATEVER = M0_FOPH_NR + 1,
+static const struct m0_fom_ops dtm0_emsg_fom_ops = {
+	.fo_fini = dtm0_fom_fini,
+	.fo_tick = dtm0_emsg_fom_tick,
+	.fo_home_locality = dtm0_fom_locality
 };
```

Comment:
Please add comments about:
What is pmsg? and what is emsg?

Reply:
Unfortunately, we (DTM team) do not have yet a well-established convention for this kind of things, so that it was kind of invented in-place. I added comments here https://github.com/Seagate/cortx-motr/pull/802/commits/d3efcfa00717f443705fe3bf9a6e3d27f5d63085#diff-7d29e2d404e16606678795ff91d5da8bc5dbc20695239029de046a0ac8bb622dR301
I hope it clarifies the meaning (DTM EXECUTE message, DTM PERSISTENT message). If it is not, please let me know, I'll add more information.

Reply:
OK